ragnarok_formats/src/model.rs

```diff
@@ -92,17 +149,32 @@ pub struct TextureCoordinateData {
 #[cfg_attr(feature = "interface", derive(korangar_interface::elements::PrototypeElement))]
 pub struct NodeData {
     pub node_name: ModelString<40>,
-    pub parent_node_name: ModelString<40>, // This is where 2.2 starts failing
-    pub texture_count: u32,
-    #[repeating(texture_count)]
+    pub parent_node_name: ModelString<40>,
+    #[version_smaller(2, 3)]
+    #[new_derive]
+    pub texture_count: Option<u32>,
+    #[repeating_option(texture_count)]
     pub texture_indices: Vec<u32>,
+    #[version_equals_or_above(2, 3)]
+    #[new_derive]
+    pub texture_name_count: Option<u32>,
+    #[repeating_option(texture_name_count)]
+    pub texture_names: Vec<ModelString<40>>,
     #[cfg_attr(feature = "interface", hidden_element)]
     pub offset_matrix: Matrix3<f32>,
-    pub translation1: Vector3<f32>,
+    #[version_smaller(2, 2)]
+    #[new_default]
+    pub translation1: Option<Vector3<f32>>,
     pub translation2: Vector3<f32>,
-    pub rotation_angle: f32,
-    pub rotation_axis: Vector3<f32>,
-    pub scale: Vector3<f32>,
+    #[version_smaller(2, 2)]
+    #[new_default]
+    pub rotation_angle: Option<f32>,
+    #[version_smaller(2, 2)]
+    #[new_default]
+    pub rotation_axis: Option<Vector3<f32>>,
+    #[version_smaller(2, 2)]
+    #[new_default]
+    pub scale: Option<Vector3<f32>>,
```

Comment:
Why are `translation1`, `rotation_angle`, `rotation_axis` and `scale` marked with `new_default`?

`new_default` is meant for fields that the user should never care about setting when creating a new instance of the type (like the signature, unknown bytes, patting, etc.).

EDIT: Actually, same for `frames_per_second`